src/plugins/dashboard/public/application/dashboard_app_empty.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import {
+  AnalyticsNoDataPageKibanaProvider,
+  AnalyticsNoDataPage,
+} from '@kbn/shared-ux-page-analytics-no-data';
+import { DataPublicPluginStart } from '@kbn/data-plugin/public';
+
+import { DashboardAppServices } from '../types';
+import { useKibana } from '../services/kibana_react';
+
+export const DashboardAppEmpty = ({ onDataViewCreated }: { onDataViewCreated: () => void }) => {
+  const {
+    services: { core, data, dataViewEditor },
+  } = useKibana<DashboardAppServices>();
+  const analyticsServices = {
+    coreStart: core as unknown as React.ComponentProps<
+      typeof AnalyticsNoDataPageKibanaProvider
+    >['coreStart'],
+    dataViews: data.dataViews,
+    dataViewEditor,
+  };
+  return (
+    <AnalyticsNoDataPageKibanaProvider {...analyticsServices}>
+      <AnalyticsNoDataPage onDataViewCreated={onDataViewCreated} />;
+    </AnalyticsNoDataPageKibanaProvider>
+  );
+};
+
+export const isDashboardAppEmpty = async (dataViews: DataPublicPluginStart['dataViews']) => {
```

Comment:
just a nit: we're trying to standardize terminology in a way that by "no data" we refer to "there is no data in ES or there are no data views". by "empty" we refer to "this app is empty", aka  there is data, but there are no dashboards yet. can we rename this to something like "noDataCheck" or "hasData" or eve "canUseDashboard"?

Reply:
Ah, I will rename all of these. Naming is important, thank you for pointing this out!